modules/script.nix

```diff
@@ -0,0 +1,57 @@
+{lib, cfg, scfg, name}:
+let
+  inherit (scfg)
+    environmentKey quoteEnvironmentValues
+    environmentVariableNamePrefix extraScript
+    environmentFile user group secretsKey secretsAreBase64;
+  inherit (lib) optionalString toUpper;
+
+  secretsPath = "${cfg.outPrefix}/${name}";
+in
+(''
+  set -euo pipefail
+  source ${environmentFile}
```

Comment:
Hmm, I don't think this is a good idea to stop using native systemd `EnvironmentFile` just to provide a workaround for launchd, I believe you can something like
```
script = ''
  source ${environmentFile}
'' + import ./script.nix { inherit lib cfg scfg name; };
```